flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala

```diff
@@ -1096,29 +1096,6 @@ object ScalarOperatorGens {
           operandTerm => s"$operandTerm.toBytes($serTerm)"
         }
 
-      // Note: SQL2003 $6.12 - casting is not allowed between boolean and numeric types.
```

Comment:
Is this comment worth to be preserved? I guess it is even wrong, because Calcite allows certain numeric types?

Reply:
If it's worth to be preserved, I guess we should place it somewhere else? Perhaps in `IntegerNumericToBooleanCastRule`?

Reply:
I just wanted to raise attention. But I think it is not valid anymore. Let's not preserve it.

Reply:
I would actually move it maybe to `LogicalTypeCasts`? So that we know that we chose to implement them instead of the SQL spec?